src/Infrastructure/Processor/Process/MediaTranslationShopware6ExportProcess.php

```diff
@@ -0,0 +1,150 @@
+        } catch (Throwable $exception) {
+            $this->exportRepository->addError($export->getId(), $exception->getMessage(), []);
+        }
+    }
+
+    /**
+     * @throws GuzzleException
+     */
+    private function ensureMediaTranslations(
+        Shopware6Channel $channel,
+        AbstractMultimedia $multimedia,
+        string $shopwareId
+    ): void {
+        //apply translations from shopware
+        $shopware6Media = new Shopware6Media($shopwareId, null);
+        $shopware6Media->setTranslations($this->getMediaTranslations($channel, $shopwareId));
+        //apply translations from ergo
+        $altTranslations = $multimedia->getAlt()->getTranslations();
+        $titleTranslations = $multimedia->getTitle()->getTranslations();
+        $languages = array_unique(array_merge(array_keys($altTranslations), array_keys($titleTranslations)));
+        $ergoTranslations = [];
+        foreach ($channel->getLanguages() as $channelLanguage) {
+            if (!in_array($channelLanguage->getCode(), $languages)) {
+                continue;
+            }
+            if ($this->languageRepository->exists($channel->getId(), $channelLanguage->getCode())) {
+                $shopwareLanguage = $this->languageRepository->load($channel->getId(), $channelLanguage->getCode());
+                Assert::notNull(
+                    $shopwareLanguage,
+                    sprintf('Expected a value other than null for product lang  %s', $channelLanguage->getCode())
+                );
+                $ergoTranslations[] = new Shopware6MediaTranslation(
+                    null,
+                    $altTranslations[$shopwareLanguage->getIso()] ?? null,
+                    $titleTranslations[$shopwareLanguage->getIso()] ?? null,
+                    $shopwareLanguage->getId()
+                );
+            }
+        }
+        $shopware6Media->updateTranslated($ergoTranslations);
+
+        if ($shopware6Media->isModified()) {
+            $this->logger->info(
+                'Updating multimedia media translations start',
+                [
+                    'multimediaId' => $multimedia->getId()->getValue(),
+                    'multimediaName' => $multimedia->getName(),
+                    'mediaId' => $shopware6Media->getId()
+                ]
+            );
+            $this->updateTranslations($channel, $shopware6Media);
+            $this->logger->info(
+                'Updating multimedia media translations end',
+                [
+                    'multimediaId' => $multimedia->getId()->getValue(),
+                    'multimediaName' => $multimedia->getName(),
+                    'mediaId' => $shopware6Media->getId()
+                ]
+            );
+        }
+    }
+
+    /**
+     * @throws GuzzleException
+     */
+    private function updateTranslations(Shopware6Channel $channel, Shopware6Media $shopware6Media): void
+    {
+        if (!empty($shopware6Media->getTranslations())) {
```

Comment:
what if translations are removed from Ergonode? We should check `isModified` instead

Reply:
It's already done in one update action - since shopware API has difficulties with  removal records completely - data market as deleted are marked as empty on check - see loop on Media Model file.